plugins/ipam/host-local/backend/disk/lock.go

```diff
@@ -16,36 +15,45 @@
 package disk
 
 import (
+	"github.com/alexflint/go-filemutex"
 	"os"
-	"syscall"
+	"path"
 )
 
 // FileLock wraps os.File to be used as a lock using flock
 type FileLock struct {
-	f *os.File
+	f *filemutex.FileMutex
 }
 
 // NewFileLock opens file/dir at path and returns unlocked FileLock object
-func NewFileLock(path string) (*FileLock, error) {
-	f, err := os.Open(path)
+func NewFileLock(lockPath string) (*FileLock, error) {
+	fi, err := os.Stat(lockPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if fi.IsDir() {
+		lockPath = path.Join(lockPath, "lock")
+	}
```

Comment:
is this line necessary for windows?  If so, maybe it deserves a unit test?

Reply:
yes it is required - added unit tests for filepath and folderpath

Reply:
LMK if you like what you see, will wait for LGTM to squash!

Reply:
squashed commits